### PR TITLE
Add failing test for PolynomialLR total_iters=0 closed-form crash

### PR DESCRIPTION
**Pull request by diputs-sudo**
This PR introduces a failing test that exposes a `ZeroDivisionError` when calling
`scheduler.step(epoch=...)` on `PolynomialLR` with `total_iters=0`.

Closed form LR computation performs:

```
t = min(epoch, total_iters) / total_iters
```

which produces a `0 / 0` division by zero crash when `total_iters == 0`.

This PR is **tests only**, following the incremental plan discussed in the RFC.
A future PR will add validation logic or canonical behavior for this edge case.

**Related Discussion**
This test is part of the broader cleanup work proposed in the RFC:
"RFC: Internal Unification of LRScheduler Semantics for Correct Resuming and Composition" #168044

### CODE REVIEW

File: test/optim/test_lrscheduler.py [hunk 1]
```diff
@@ -2129,7 +2129,7 @@ def test_reduce_lr_on_plateau_state_dict(self):
             self.opt, mode="max", factor=0.5, patience=10
         )
         scheduler_copy.load_state_dict(scheduler.state_dict())
-        for key in scheduler.__dict__:
+        for key in scheduler.__dict__.keys():
```
Comment (cyyever) [inline test/optim/test_lrscheduler.py]: Why `keys()`?

**Review by cyyever — CHANGES_REQUESTED** [review]

File: test/optim/test_lrscheduler.py (hunk 1 above)
Comment (diputs-sudo) [reply test/optim/test_lrscheduler.py]: This wasn’t an intentional style change on my part. When I originally made the edit, the surrounding code already used this form, and by the time the branch was pushed/rebased the file formatting had changed, so the diff ended up reflecting `.keys()` here as well.

There’s no semantic intent behind it, happy to adjust if you’d prefer a different style.
